pkg/TIPA.py

```
@@ -164,587 +165,498 @@ def __init__(self, img, mouse_pos, display_info_init, gui_queue):
+
+            a_list = search_results.find_all("a", href=True)
+            if not a_list:
+                raise Exception("No links found in search results.")
+
+            for a in a_list:
+                if site in a["href"]:
+                    return f"https://google.com{a['href']}"
+
+            return f"https://google.com{a_list[0]['href']}" if a_list else None
+
+        except Exception as e:
```

Ruff (0.8.0)

Do not catch blind exception: `Exception`

(BLE001)